File: src/main/java/com/back/mozu/domain/customer/repository/CustomerRepository.java
package com.back.mozu.domain.customer.repository;

import com.back.mozu.domain.customer.entity.Customer;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface CustomerRepository extends JpaRepository<Customer, UUID> {
    Optional<Customer> findByEmail(String email);
    Optional<Customer> findByProviderId(String providerId);
    boolean existsByEmail(String email);
}

File: src/main/java/com/back/mozu/domain/customer/repository/CustomerRepository.kt
package com.back.mozu.domain.customer.repository

import com.back.mozu.domain.customer.entity.Customer
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface CustomerRepository : JpaRepository<Customer, UUID> {
    fun findByEmail(email: String): Customer?

    fun findByEmailOrNull(email: String): Customer? = findByEmail(email)

    fun findByProviderId(providerId: String): Customer?

    fun existsByEmail(email: String): Boolean
}